src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -86,12 +91,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     List<Event> calendarEvents = calendarClientAdapter.getAcceptedEventsInTimerange(startDateTime, endDateTime);
 
-
-
     // Schedules
-    List<Task> scheduledTasks = Scheduler.scheduleInRange(
-        calendarEvents, tasksToSchedule, timeZone, startDate, endDate);
+    Scheduler scheduler = new Scheduler(calendarEvents, tasksToSchedule, timeZone);
+    List<ExtendedTask> scheduledExtendedTasks = scheduler.scheduleInRange(startDate, endDate);
 
+    List<Task> scheduledTasks = scheduledExtendedTasks.stream().map(ExtendedTask::getTask).collect(Collectors.toList());
     // Updates Tasks and Calendar
     tasksClientAdapter.updateTasks(tasksListId, scheduledTasks);
     calendarClientAdapter.insertEventsToPrimary(
```

Comment:
Add a TODO here that events should be created based on `scheduledExtendedTasks` so that their duration can be read.